systempath.d

```diff
@@ -0,0 +1,126 @@
+import std.stdio;
+import std.process;
+import std.string : startsWith, strip, indexOf;
+import std.array : replace;
+import core.vararg;
+
+// On Windows the following can happen:
+//
+// The user has installed git normally via Chocolately or MSI package
+// so it can be used in cmd.exe. The user also has git installed
+// as part of Cygwin via the Cygwin package manager.
+// 
+// Depending on which one is called the output of the following command 
+// to get the project's root folder
+//
+//    auto rootFinder = execute(["git", "rev-parse", "--show-toplevel"]);
+//
+// is either C:\path\to\repro or /path/to/repro.
+//
+// If the cmd.exe version is called within a Cygwin terminal or vice-versa
+// then all kinds of things go wrong with file and directory access.
+//
+// The problem is that D's standard library uses the 
+//    version(Windows) { ... Win32 API calls ... }
+//    version(Posix) { ... Posix calls ... }
+// compiler directives to provide file access and read/write files.
+// So internally it uses the Win32 API on Windows which expects C:\path style
+// paths, unlike what Cygwin's git version gives us, which is /path style paths.
+//
+// So the file reading comes down crashing.
+//
+// If I compile in Cyginw then everything works as expected?
+// No, unfortunately not, if the D compiler is installed using the MSI package
+// (that is, it is not a Cygwin package). Not sure what would happen if you 
+// installed D via a Cygwin package or even compile it from source in Cygwin though.
+// Feel free to investigate that route.
+//
+// The solution for this case (Cygwin git, compiled promptoglyph-vcs.exe with 
+// dmd.exe installed via non-cygwin-package, executing within Cygwin) is to
+// convert the Unix style path to a Windows style one with the handy cygpath
+// tool.
+//
+version (Windows) version = PathFixNeeded;
+version (Cygwin) version = PathFixNeeded;
+
+private:
+
+bool isCygWinEnv = false;
```

Comment:
Can we move these two declarations into `version (PathFixNeeded)`? No sense generating them if we're in Posix land.